GVFS/GVFS.Service/GVFSMountProcess.Mac.cs

```diff
@@ -1,31 +1,41 @@
 ﻿using GVFS.Common;
-using GVFS.Common.Tracing;
-using GVFS.Service.Handlers;
-using System;
+using GVFS.Common.Tracing;
+using System.Diagnostics;
+using System.IO;
 
 namespace GVFS.Service
 {
-    public class GVFSMountProcess : IDisposable
+    public class GVFSMountProcess : IRepoMounter
     {
-        private const string ParamPrefix = "--";
+        private const string ExecutablePath = "/bin/launchctl";
 
-        private readonly ITracer tracer;
+        private MountLauncher processLauncher;
+        private GVFSPlatform platform;
+        private ITracer tracer;
 
-        public GVFSMountProcess(ITracer tracer, int sessionId)
+        public GVFSMountProcess(ITracer tracer, MountLauncher processLauncher = null, GVFSPlatform platform = null)

```

Comment:
Passing `GVFSPlatform` by parameter looks a bit strange to me.  An alternative approach to consider: we could add a helper method to `GVFSPlatform` for swapping the current platform which is only intended to be use for unit tests.

_Example_

```
        /// <summary>
        /// Unit Test Only: Replaces the registered platform.
        /// </summary>
        public static void ReplaceRegisteredPlatform(GVFSPlatform platform)
        {
            GVFSPlatform.Instance = platform;
        }
```

And then in the unit test:
```
            GVFSPlatform currentPlatform = GVFSPlatform.Instance;
            try
            {
                GVFSPlatform.ReplaceRegisteredPlatform(this.gvfsPlatformMock.Object);
                GVFSMountProcess mountProcess = new GVFSMountProcess(this.tracer, mountLauncherMock.Object);
                mountProcess.MountRepository(ExpectedActiveRepoPath, ExpectedActiveUserId);

                mountLauncherMock.VerifyAll();
            }
            finally
            {
                GVFSPlatform.ReplaceRegisteredPlatform(currentPlatform);
            }
```
(If this became a common pattern we could have SetUp/Teardown methods that record/restore the platform for each test and avoid adding the `try`/`finally`)

Thoughts?

cc: @jamill, @mjcheetham, @nickgra for their thoughts as well.

Reply:
I like your proposed suggestion, although I'm open to anything that doesn't pass `GVFSPlatform` by parameter.

Reply:
Based on our previous offline discussions, I think the longer term idea is to pass dependencies explicitly and not rely on a static instance variable (possibly via a context object or something similar).

I think @wilbaker's suggestion is one way to get this under test. An alternative is to require these properties to always be passed into the constructor, which has the property that the dependencies of this class are more explicit, but is not consistent with how we do this elsewhere...

At this point, I might lean towards @wilbaker's suggestion, but I personally wouldn't be opposed to always passing this in via the constructor.

/cc @kewillford

Reply:
Usually you want to limit the scope of the dependencies you are passing in via the constructor to only what you need instead of the whole "kitchen sink". So in this case we shouldn't really be passing in the `GVFSPlatform` since the class is only using the `Constants` it should be the `GVFSPlatformConstants` or maybe some smaller interface that only has what you need.

That being said it looks like we already have this functionality here:
```
        [OneTimeSetUp]
        public void SetUp()
        {
            GVFSPlatform.Register(new MockPlatform());
        }
```

so I think we should stick with this model for now and not have the `GVFSPlatform` passed in via constructors.  You can also call the `GVFSPlatform.Register` in your individual tests if needed to test something more specific about the `GVFSPlatform`.

Reply:
Thank you all for the inputs. Agree with the idea that passing around an object that holds global state information should not be necessary.

Updated the `service` tests to use `MockPlatform`. `MockPlatform` is not a `Moq` mock, so `Setup` is not available and I had to add a `MockCurrentUser` property to hold a fake current user.